skll/logutils.py

```diff
@@ -53,5 +58,38 @@ def get_skll_logger(name, filepath=None, log_level=logging.INFO):
             file_handler.setLevel(log_level)
             logger.addHandler(file_handler)
 
+    # Make method for sending sklearn-specific warnings to the
+    # logger that can be used to replace warnings.showwarning
+    def send_sklearn_warnings_to_log(message, category, filename, lineno,
+                                     file=None, line=None):
+        if SKLEARN_WARNINGS_RE.search(filename):
+            logger.warning('{}:{}: {}:{}'.format(filename,
+                                                 lineno,
+                                                 category.__name__,
+                                                 message))
+        else:
+            orig_showwarning(message, category, filename, lineno,
+                             file=file, line=line)
+    warnings.showwarning = send_sklearn_warnings_to_log
+
     # return the logger instance
     return logger
+
+
+def close_and_remove_logger_handlers(logger):
```

Comment:
Thanks for adding this functions and catching the issue of unclosed file handlers.